src/main/java/com/coherent/training/api/kapitsa/base/BaseClientObject.java

```diff
@@ -72,12 +73,12 @@ private HttpUriRequest setRequest(String endpoint, String method, Map<String, St
         return requestBuilder.build();
     }
 
-    private HttpUriRequest setRequestWithParams(String endpoint, String method, Map<String, String> headersMap, HttpEntity entity, NameValuePair nvp) {
+    private HttpUriRequest setRequestWithParams(String endpoint, String method, Map<String, String> headersMap, NameValuePair[] nvp, HttpEntity... entity) {
         requestBuilder = RequestBuilder.create(method)
                 .setUri(endpoint)
-                .addParameter(nvp)
-                .setEntity(entity);
+                .addParameters(nvp);
 
+        if (entity.length > 0) requestBuilder.setEntity(entity[0]);
```

Comment:
even if you have one instruction as 'if' body please wrap it into curly brackets